scheduler-server/sql/scheduler/tables/scheduling_specification_goals.sql

```diff
@@ -1,7 +1,10 @@
 create table scheduling_specification_goals (
   specification_id integer not null,
   goal_id integer not null,
-  priority integer not null constraint non_negative_specification_goal_priority check (priority >= 0),
+  priority integer
+    not null
```

Comment:
Hmm, how does `not null default null` work? Seems counter intuitive

Reply:
+1

Reply:
As long as the null is replaced with a number by the trigger before its inserted, thats fine. I considered making the default -1 (which would also violate a constraint if not changed), but I figured if something went wrong in the trigger and it wasn't updated, it would be easier to debug if it was null.

Reply:
I would have liked to do a autoincrement-related solution, but we have multiple lists interleaved in the tables (one list per spec or template), so autoincrement or SERIAL wouldn't work.

Reply:
I've added a comment explaining this, or I don't mind doing it a different way if you want.

Reply:
Thanks for the comment :+1: I think this is fine - the timing of the constraints and the triggers is a bit nuanced, but the comment helps elucidate that.

So - my current understanding is that the order of events is:
1. INSERT statement comes in, and it's missing the priority field
2. That priority field is set to null
3. The trigger replaces the null with a number
4. The NOT NULL constraint is checked - if we did our job right, this should never fail